src/Analyser/TypeSpecifier.php

```diff
@@ -180,15 +180,13 @@ public function specifyTypesInCondition(
 				return $this->create($exprNode, new ObjectWithoutClassType(), $context, false, $scope, $rootExpr);
 			}
 		} elseif ($expr instanceof Node\Expr\BinaryOp\Identical) {
+			$binaryExpressionSpecifiedType = null;
 			$expressions = $this->findTypeExpressionsFromBinaryOperation($scope, $expr);
 			if ($expressions !== null) {
 				$exprNode = $expressions[0];
 				$constantType = $expressions[1];
 
-				$specifiedType = $this->specifyTypesForConstantBinaryExpression($exprNode, $constantType, $context, $scope, $rootExpr);
```

Comment:
this case might specify types besides `$exprNode`, therefore remember this types and later on return them in union with other known things within the current scope instead of ealry returning